nemo/collections/nlp/models/language_modeling/megatron/gpt_layer_modelopt_spec.py

```diff
@@ -57,7 +55,8 @@ def get_gpt_layer_modelopt_spec(num_experts: Optional[int] = None) -> ModuleSpec
     if not HAVE_MEGATRON_CORE:
         raise IMPORT_ERROR
 
-    mlp = _get_mlp_module_spec(num_experts=num_experts)
+    mlp = _get_mlp_module_spec(use_te=False, num_experts=num_experts, moe_grouped_gemm=False, fp8=False)
```

Comment:
```suggestion
    mlp = _get_mlp_module_spec(use_te=False, num_experts=num_experts, moe_grouped_gemm=False)
```

See deprecation warning.

Generally, I think it's a bit sub-optimal to use a private function. But to move fast, let's approve this PR and work on a refactor of Mcore separately. Once Mcore provides a public version of `get_mlp_module_spec`, we should switch

Reply:
Right, totally agreed. In long term we're planning to import the "modelopt" spec right away from [megatron/core/inference/modelopt_support/gpt/model_specs.py](https://gitlab-master.nvidia.com/ADLR/megatron-lm/-/blob/main/megatron/core/inference/modelopt_support/gpt/model_specs.py?ref_type=heads) (which btw. also imports this private function). There are however CI tests needed for this spec in Megatron to make sure stays functional as we observed some issues with it in past. But we hope to migrate to using it soon.